src/GitHub.Api/SimpleApiClientFactory.cs

```diff
@@ -28,10 +28,6 @@ public SimpleApiClientFactory(IProgram program, Lazy<IEnterpriseProbeTask> enter
 
         public ISimpleApiClient Create(UriString repositoryUrl)
         {
-            var contains = cache.ContainsKey(repositoryUrl);
-            if (contains)
-                return cache[repositoryUrl];
-
```

Comment:
Yeah, I'm fine with that behavior because creating them is cheap. Especially if it's discarded. Locks are less cheap.